extensions/amp-bind/0.1/test/test-bind-integration.js

```diff
@@ -475,4 +487,45 @@ describe.configure().retryOnSaucelabs().run('amp-bind', function() {
       });
     });
   });
+
+  describe('amp-state', () => {
+    it('should not loop infinitely if updates change its src binding', () => {
+      // Changes amp-state's src to ...amp-state-src2
+      const changeAmpStateSrcButton =
+          fixture.doc.getElementById('ampStateSrcButton');
+      const triggerBindApplicationButton =
+          fixture.doc.getElementById('triggerBindApplicationButton');
+      const ampState = fixture.doc.getElementById('ampState');
+      const batchedXhr = batchedXhrFor(fixture.win);
+      // Stub XHR for endpoint ...amp-state-src2 that returns state that would
+      // redirect the amp-state's src back to the original src, amp-state-src.
+      sandbox.stub(batchedXhr, 'fetchJson').returns(Promise.resolve({
+        ampStateSrc: 'https://www.google.com/bind/first/source',
+      }));
+      const spy =
+          sandbox.spy(ampState.implementation_, 'mutatedAttributesCallback');
```

Comment:
Instead of spying on an internal method which is weird for an integration test, can we use externally visible state as a proxy? I.e. the value of the `src` attribute.